pkg/envoy/lds/listener_test.go

```diff
@@ -3,62 +3,21 @@ package lds
 import (
 	"testing"
 
-	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
 	xds_type "github.com/envoyproxy/go-control-plane/envoy/type/v3"
-	"github.com/golang/mock/gomock"
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/assert"
 
-	configv1alpha2 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
-
-	"github.com/openservicemesh/osm/pkg/catalog"
-	"github.com/openservicemesh/osm/pkg/configurator"
-	"github.com/openservicemesh/osm/pkg/constants"
-	"github.com/openservicemesh/osm/pkg/envoy"
-	"github.com/openservicemesh/osm/pkg/identity"
-	"github.com/openservicemesh/osm/pkg/service"
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
-// Tests TestGetFilterForService checks that a proper filter type is properly returned
-// for given config parameters and service
-
-var _ = Describe("Construct inbound listeners", func() {
-	var (
-		mockCtrl         *gomock.Controller
-		mockConfigurator *configurator.MockConfigurator
-	)
-
-	mockCtrl = gomock.NewController(GinkgoT())
-	mockConfigurator = configurator.NewMockConfigurator(mockCtrl)
+func TestBuildPrometheusListener(t *testing.T) {
+	a := assert.New(t)
 
-	mockConfigurator.EXPECT().IsTracingEnabled().Return(false).AnyTimes()
-	mockConfigurator.EXPECT().GetTracingHost().Return(constants.DefaultTracingHost).AnyTimes()
-	mockConfigurator.EXPECT().GetTracingPort().Return(constants.DefaultTracingPort).AnyTimes()
-
-	Context("Test creation of inbound listener", func() {
-		It("Tests the inbound listener config", func() {
-			listener := newInboundListener()
-			Expect(listener.Address).To(Equal(envoy.GetAddress(constants.WildcardIPAddr, constants.EnvoyInboundListenerPort)))
-			Expect(listener.AccessLog).NotTo(BeEmpty())
-			Expect(len(listener.ListenerFilters)).To(Equal(2)) // TlsInspector, OriginalDestination listener filter
-			Expect(listener.ListenerFilters[0].Name).To(Equal(envoy.TLSInspectorFilterName))
-			Expect(listener.TrafficDirection).To(Equal(xds_core.TrafficDirection_INBOUND))
-		})
-	})
-
-	Context("Test creation of Prometheus listener", func() {
-		It("Tests the Prometheus listener config", func() {
-			connManager := getPrometheusConnectionManager()
-			listener, _ := buildPrometheusListener(connManager)
-			Expect(listener.Address).To(Equal(envoy.GetAddress(constants.WildcardIPAddr, constants.EnvoyPrometheusInboundListenerPort)))
-			Expect(len(listener.ListenerFilters)).To(Equal(0)) //  no listener filters
-			Expect(listener.TrafficDirection).To(Equal(xds_core.TrafficDirection_INBOUND))
-		})
-	})
-})
+	connManager := getPrometheusConnectionManager()
+	listener, err := buildPrometheusListener(connManager)
+	a.NotNil(listener)
```

Comment:
should there be more checks here?  seems like the previous test had more checks on the listener such as: 

```
Expect(listener.Address).To(Equal(envoy.GetAddress(constants.WildcardIPAddr, constants.EnvoyPrometheusInboundListenerPort)))
```

Reply:
I considered it but it's unnecessary testing the expected config by populating the fields the same way as the actual config. We already have e2e tests for this, and this isn't going to increase code coverage anyway.